onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -124,9 +127,131 @@ def _parse_line(line):
+        parameter model: loaded FP32 ONNX model to quantize
+        return: augmented ONNX model
+    '''
+    added_nodes = []
+    added_outputs = []
+    for node in model.graph.node:
+        if node.op_type in selected_types:
+            input_name = node.name
+            # Adding ReduceMin nodes
+            reduce_min_name = input_name + '_ReduceMin'
+            reduce_min_node = onnx.helper.make_node('ReduceMin', [input_name + ':0'],
+                            [reduce_min_name + ':0'], reduce_min_name, keepdims=0)
+            added_nodes.append(reduce_min_node)
+            added_outputs.append(helper.make_tensor_value_info(reduce_min_node.name + ':0', TensorProto.FLOAT, ()))
+            # Adding ReduceMax nodes
+            reduce_max_name = input_name + '_ReduceMax'
+            reduce_max_node = onnx.helper.make_node('ReduceMax', [input_name + ':0'],
+                            [reduce_max_name + ':0'], reduce_max_name, keepdims=0)
+            added_nodes.append(reduce_max_node)
+            added_outputs.append(helper.make_tensor_value_info(reduce_max_node.name + ':0', TensorProto.FLOAT, ()))
+    model.graph.node.extend(added_nodes)
+    model.graph.output.extend(added_outputs)
+    return model
+
+# Reading user input (images)
+def load_and_resize_image(image_filepath, height, width):
+    '''
+    Resizes image to  to NCHW format
+        parameter image_filepath: path to image files
+        parameter height: image height in pixels
+        parameter width: image width in pixels
+        return: matrix characterizing image
+    '''
+    pillow_img = Image.open(image_filepath).resize((width, height))
+    input_data = np.float32(pillow_img)
+    nhwc_data = np.expand_dims(input_data, axis=0)
+    nchw_data = nhwc_data.transpose(0, 3, 1, 2) # ONNX Runtime standard
+    return nchw_data
+
+def load_batch(images_folder, height, width, size_limit=30):
+    '''
+    Loads a batch of images
+    parameter images_folder: path to folder storing images
+    parameter height: image height in pixels
+    parameter width: image width in pixels
+    parameter size_limit: number of images used to run inference
+    return: list of matrices characterizing multiple images
+    '''
+    image_names = os.listdir(images_folder)
+    if len(image_names) >= size_limit:
+        batch_filenames = [image_names[i] for i in range(size_limit)]
+    else:
+        batch_filenames = image_names
+    unconcatenated_batch_data = []
+    for image_name in batch_filenames:
+        image_filepath = images_folder + '/' + image_name
+        nchw_data = load_and_resize_image(image_filepath, height, width)
+        unconcatenated_batch_data.append(nchw_data)
+    batch_data = np.concatenate(np.expand_dims(unconcatenated_batch_data, axis=0), axis=0)
+    return batch_data
+
+# Using augmented outputs to generate inputs to quantize.py
+def get_intermediate_outputs(augmented_model_path, inputs, average_mode='naive'):
```

Comment:
add a better name instead of average mode...

Reply:
Changed!